From 48207476214132a07de61eefbaf6cbefceaa1fa5 Mon Sep 17 00:00:00 2001
From: Rex Zhu <Rex.Zhu@amd.com>
Date: Thu, 17 Dec 2015 14:20:06 +0800
Subject: [PATCH 0959/1050] drm/amd/powerplay: check whether need to enable
 thermal control.

In I+A platform(skylake), it is controlled by intel.

Reviewed-by: Alex Deucher <alexander.deucher@amd.com>

Change-Id: I1030bef2af7e1740e8b6077d2f9b6a8f3da0b878

Signed-off-by: Rex Zhu <Rex.Zhu@amd.com>
---
 drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c b/drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c
index 0a03f79..707664d 100644
--- a/drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c
+++ b/drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c
@@ -418,10 +418,17 @@ restart_search:
 int pem_task_initialize_thermal_controller(struct pp_eventmgr *eventmgr, struct pem_event_data *event_data)
 {
 	struct PP_TemperatureRange range;
+
 	range.max = TEMP_RANGE_MAX;
 	range.min = TEMP_RANGE_MIN;
 
-	return phm_start_thermal_controller(eventmgr->hwmgr, &range);
+	if (eventmgr == NULL || eventmgr->platform_descriptor)
+		return -EINVAL;
+
+	if (phm_cap_enabled(eventmgr->platform_descriptor->platformCaps, PHM_PlatformCaps_ThermalController))
+		return phm_start_thermal_controller(eventmgr->hwmgr, &range);
+
+	return 0;
 }
 
 int pem_task_uninitialize_thermal_controller(struct pp_eventmgr *eventmgr, struct pem_event_data *event_data)
-- 
1.9.1

